src/components/search/data/hooks.js

```diff
@@ -137,9 +137,13 @@ export const useDefaultSearchFilters = ({ enterpriseConfig, subscriptionPlan, of
 
   const filters = useMemo(
     () => {
+      const customerCatalogFilter = `enterprise_customer_uuids:${enterpriseConfig.uuid}`;
       if (showAllCatalogs) {
         // show all enterprise catalogs
-        return `enterprise_customer_uuids:${enterpriseConfig.uuid}`;
+        if (!subscriptionPlan) {
+          return customerCatalogFilter;
+        }
+        return `enterprise_catalog_uuids:${subscriptionPlan.enterpriseCatalogUuid} OR ${customerCatalogFilter}`;
```

Comment:
I'm still a bit confused why we need to filter on both the catalog and the customer uuids here; why won't just the customer uuid filter work? The subs catalog would still come back with the customer uuid filter alone.

Reply:
I thought it would work that way, and when I tested locally I thought it was working, but I checked on stage with subscription and codes and it wasn't working without this. :(